packages/host/app/services/store.ts

```diff
@@ -1216,6 +1193,16 @@ function processCardError(
   }
 }
 
+function needsServerStateMerge(
+  instance: CardDef,
+  serverState: SingleCardDocument,
+): boolean {
+  return (
+    instance.id === serverState.data.id &&
+    isEqual(instance[meta], serverState.data.meta)
```

Comment:
Is there some kind of version in meta? Why is this check meaningful?

Reply:
probably we can be even more precise here, in that we don't want to merge server state when the polymorphic field info changes. the store already knows about that...

Reply:
lol, i named this function backwards from what it is doing